### ISSUE TITLE
Invalid queries on Django 1.8+

### ISSUE THREAD

**Issue by Stranger6667** [django-money/django-money#166]
As I've mentioned in #165 there is a bug on Django 1.8+.
Q objects generates different queries then regular kwargs passed to manager.filter() on Django 1.8+
It doesn't add `currency` part to WHERE clause. I've skipped equal parts (Python 2.7, Django 1.9)
```python
>>> print str(ModelWithTwoMoneyFields.objects.filter(Q(amount1__gt=F('amount2'))).query)
... WHERE "testapp_modelwithtwomoneyfields"."amount1" > ("testapp_modelwithtwomoneyfields"."amount2")
>>> print str(ModelWithTwoMoneyFields.objects.filter(amount1__gt=F('amount2')).query)
... WHERE ("testapp_modelwithtwomoneyfields"."amount1" > ("testapp_modelwithtwomoneyfields"."amount2") AND "testapp_modelwithtwomoneyfields"."amount1_currency" = ("testapp_modelwithtwomoneyfields"."amount2_currency"))
```
On Django versions prior to 1.8 queries are logically equal.
I'll create new PR for this issue after #165 will be merged to master (I hope it will :))
